lib/codegen/fromcto/rust/rustvisitor.js

```diff
@@ -225,6 +233,39 @@ class RustVisitor {
             }
         }
         parameters.fileWriter.writeLine(1, ')]');
+
+        if (!ModelUtil.isPrimitiveType(field.type) && ModelUtil.isMap(field)) {
+            const mapDeclaration = field.getModelFile().getType(field.getType());
+            const mapKeyType     = mapDeclaration.getKey().getType();
+            const mapValueType   = mapDeclaration.getValue().getType();
+
+            let rustKeyType;
+            let rustValueType;
+
+            // Key
+            if (ModelUtil.isPrimitiveType(mapKeyType)) {
+                rustKeyType = this.toRustType(mapKeyType);
```

Comment:
The comment on the scalars goes for primitive types too. https://github.com/accordproject/concerto/blob/5b72c1951c4f1546cbc29a8a18e89d516efa74e3/packages/concerto-core/lib/modelutil.js#L159-L162

Reply:
@ ekarademir any chance you can get this PR over the line so we can merge?

Reply:
> @ ekarademir any chance you can get this PR over the line so we can merge?

Definitely!

Reply:
@dselman  #87 has been merged. I couldn't only create a fork so I couldn't update this branch.